llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -1125,6 +1134,7 @@ GCNTargetMachine::getSubtargetImpl(const Function &F) const {
   }
 
   I->setScalarizeGlobalBehavior(ScalarizeGlobal);
+  I->setOOBMode(getOOBModeFromModule(F.getParent()));
```

Comment:
The subtarget should not be modified based on current compilation state. The subtarget should be treated as a universal constant. If you really want to have a separate subtarget based on the module flag, you should add the module flag as another key in the subtarget map